### PR TITLE
feat: move metadata list to annotation

### PR DESCRIPTION
**Pull request by weixiao-huang**
### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/base/runtime.go [hunk 1]
```diff
@@ -397,7 +406,7 @@ func GetRuntimeInfo(client client.Client, name, namespace string) (runtimeInfo R
 		if err != nil {
 			return runtimeInfo, err
 		}
-		runtimeInfo, err = BuildRuntimeInfo(name, namespace, common.EFCRuntime, datav1alpha1.TieredStore{}, WithMetadataList(efcRuntime.Spec.MetadataList))
+		runtimeInfo, err = BuildRuntimeInfo(name, namespace, common.EFCRuntime, datav1alpha1.TieredStore{}, WithMetadataList(GetMetadataListFromAnnotation(efcRuntime)))
```
Comment (TrafalgarZZZ) [inline pkg/ddc/base/runtime.go:409]: Please also modify the following funcs to make metadata list work when creating runtimes:

https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/alluxio/runtime_info.go#L33
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/efc/runtime_info.go
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/goosefs/runtime_info.go
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/jindofsx/runtime_info.go#L32
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/juicefs/runtime_info.go#L33
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/thin/runtime_info.go#L33

**Review by TrafalgarZZZ — CHANGES_REQUESTED** [review]

File: pkg/ddc/base/runtime.go [hunk 2]
```diff
@@ -171,6 +173,13 @@ func BuildRuntimeInfo(name string,
 
 type RuntimeInfoOption func(info *RuntimeInfo)
 
+func GetMetadataListFromAnnotation(accessor metav1.ObjectMetaAccessor) (ret []datav1alpha1.Metadata) {
+	annotations := accessor.GetObjectMeta().GetAnnotations()
```
Comment (cheyang) [inline pkg/ddc/base/runtime.go:177]: I think `json.Unmarshal()` can have some CPU overheads, especially if it is called frequently or used to parse large amounts of data. I have the following suggestions:

1. Checking if the specified metadata is empty. If it is, there is no need to execute the further parsing operation.
2. Checking if  the presence of the data.fluid.io/metadataList key exist. If it isn't,  there is no need to execute the further parsing operation.
3. Checking the type of metadata. If the type is incorrect, the parsing operation cannot be completed. In this case, an error message should be logging at least. if the metadata is of the wrong type, no one will be aware of this.

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/ddc/base/runtime.go (hunk 2 above)
Comment (weixiao-huang) [reply pkg/ddc/base/runtime.go:177]: I'll do the first and second steps, but in the third step, which log should I use?

**Review by cheyang — APPROVED** [review]

**Comment by cheyang** [comment]
@TrafalgarZZZ Please take a look when you are back. No urgent. Thanks.

**Review by TrafalgarZZZ — APPROVED** [review]
/lgtm
Nice work for supporting this!

**Pull request merged**